packages/react/src/components/Toggletip/__tests__/Toggletip-test.js

```diff
@@ -4,23 +4,183 @@
+      <Toggletip data-testid="toggletip" defaultOpen>
+        <ToggletipButton label="Show information">test</ToggletipButton>
+      </Toggletip>
+    );
+    userEvent.type(screen.getByRole('button'), 'space');
+    expect(screen.getByRole('button')).toHaveAttribute(
+      'aria-expanded',
+      'false'
+    );
+    expect(container.firstChild).not.toHaveClass('cds--toggletip--open');
+    expect(container.firstChild).not.toHaveClass('cds--popover--open');
+  });
+  it('should close if an element outside of the toggletip is clicked', () => {
+    const { container } = render(
+      <Toggletip data-testid="toggletip" defaultOpen>
+        <ToggletipButton label="Show information">test</ToggletipButton>
+      </Toggletip>
+    );
+    fireEvent.click(document);
+    expect(screen.getByRole('button')).toHaveAttribute(
+      'aria-expanded',
+      'false'
+    );
+    expect(container.firstChild).not.toHaveClass('cds--toggletip--open');
+    expect(container.firstChild).not.toHaveClass('cds--popover--open');
+  });
 
   describe('Component API', () => {
-    it.todo('should support custom elements with the `as` prop');
-    it.todo('should support a custom class name with the `className` prop');
-    it.todo('should support different alignments with the `align` prop');
-    it.todo('should initially be open if `defaultOpen` is set to true');
+    it('should support custom elements with the `as` prop', () => {
+      const CustomComponent = forwardRef((props, ref) => {
+        return <div data-testid="custom-component" ref={ref} {...props} />;
+      });
+
+      render(<Toggletip as={CustomComponent}>test</Toggletip>);
+
+      expect(screen.getByTestId('custom-component')).toBeInTheDocument();
+    });
+
+    it('should support a custom class name with the `className` prop', () => {
+      const { container } = render(
+        <Toggletip data-testid="toggletip" className="custom-class">
+          <span>test</span>
+        </Toggletip>
+      );
+
+      expect(container.firstChild).toHaveClass('custom-class');
+    });
+    it.each([
+      ['left', 'cds--popover--left'],
+      ['right', 'cds--popover--right'],
+      ['top', 'cds--popover--top'],
+      ['bottom', 'cds--popover--bottom'],
+      ['top-left', 'cds--popover--top-left'],
+      ['top-right', 'cds--popover--top-right'],
+      ['bottom-left', 'cds--popover--bottom-left'],
+      ['bottom-right', 'cds--popover--bottom-right'],
+      ['left-top', 'cds--popover--left-top'],
+      ['left-top', 'cds--popover--left-top'],
+      ['right-top', 'cds--popover--right-top'],
+      ['right-bottom', 'cds--popover--right-bottom'],
+    ])(
+      'should support different alignments with the `align` prop',
+      (align, className) => {
+        const { container } = render(
+          <Toggletip data-testid="toggletip" align={align}>
+            <span>test</span>
+          </Toggletip>
+        );
+
+        expect(container.firstChild).toHaveClass(className);
+      }
+    );
```

Comment:
Constructing the className programmatically would remove the need to specify all the classNames in the array
```suggestion
    it('should support a custom class name with the `className` prop', () => {
      const { container } = render(
        <Toggletip data-testid="toggletip" className="custom-class">
          <span>test</span>
        </Toggletip>
      );

      expect(container.firstChild).toHaveClass('custom-class');
    });
    it.each([
      ['left', 'cds--popover--left'],
      ['right', 'cds--popover--right'],
      ['top', 'cds--popover--top'],
      ['bottom', 'cds--popover--bottom'],
      ['top-left', 'cds--popover--top-left'],
      ['top-right', 'cds--popover--top-right'],
      ['bottom-left', 'cds--popover--bottom-left'],
      ['bottom-right', 'cds--popover--bottom-right'],
      ['left-top', 'cds--popover--left-top'],
      ['left-top', 'cds--popover--left-top'],
      ['right-top', 'cds--popover--right-top'],
      ['right-bottom', 'cds--popover--right-bottom'],
    ])(
      'should support different alignments with the `align` prop',
      (align) => {
        const { container } = render(
          <Toggletip data-testid="toggletip" align={align}>
            <span>test</span>
          </Toggletip>
        );

        expect(container.firstChild).toHaveClass(`cds--popover--${align}`);
      }
    );
```